### PR TITLE
VEN-1405 | New Boat Information view & link to creating new boats view

### PR DESCRIPTION
**Pull request by Jokinen**
## Description :sparkles:

Allows user to use an existing boat when they make an application.

Path 1: User does not have boat
- User is redirected to CREATE_BOAT phase

Path 2: User has one boat
- User is redirected to VERIFY_BOAT phase
- User can select to create a new boat

Path 3: User has more than a single boat
- User is redirected to SELECT_BOAT phase
- User can choose to create a new boat or select an existing boat

When the form is submitted, the values are normalised depending on the presence of a `boatId`. If there's a `boatId` other boat related fields are removed. The fields are not removed earlier because some form phases depend on them--if the user were to go back enough steps the may have to re-input some data.

## Screenshots :camera_flash:

Screenshot 2021-11-23 at 14 18 41
Screenshot 2021-11-23 at 14 18 54

### CODE REVIEW

File: src/features/berth/berthFormPage/boatDetails/BoatDetailsSelectBoat.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  const { t } = useTranslation();
+
+  return (
+    <BoatDetailsPhase>
+      <BoatDetailsPhase.Section>
+        <HDSSelect
+          id="boatId"
+          name="boatId"
+          label={t('form.registered.field.boat_id.label')}
+          options={boats.map((boat: any) => ({ value: boat.id, label: boat.name }))}
```
Comment (Nizar-Rahme) [inline src/features/berth/berthFormPage/boatDetails/BoatDetailsSelectBoat.tsx:28]: `boats` is already typed, why do you have to suppress it here?

File: src/features/berth/berthFormPage/boatDetails/BoatDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  // into this phase automatically if they only have a single
+  // boat.
+  VERIFY_BOAT = 3,
+  // User can add a new boat in case they do not have
+  // a boat or the boat they are making an application for
+  // is not added yet.
+  CREATE_BOAT = 4,
+}
+
+function getInitialPhase(boats?: any[]) {
```
Comment (Nizar-Rahme) [inline src/features/berth/berthFormPage/boatDetails/BoatDetails.tsx:25]:
```suggestion
function getInitialPhase(boats?: Boat[]) {
```
If the type doesn't matter, I would rather use `unknown`

**Comment by Jokinen** [comment]
Good comments, I'll resolve tomorrow 👍

**Review by Nizar-Rahme — APPROVED** [review]
Nicely done 👍

**Pull request merged**